### PR TITLE
Explain adding hardware / make targets

### PR DESCRIPTION
**Pull request by TheSmartGerman**
More explenation how to get fast/easy hardware pinnigs. Add make targets from make help.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -14,7 +14,9 @@ The final hex file is named `ethersex.hex`.
 
 How to add a new hardware pinning
 =================================
-Use the script at `scripts/add-hardware` to add a new pinning.
+* Make `make menuconfig` and configure hardware / select features
+* Save configuration as .config file
+* Use the script at `scripts/add-hardware` to add a new pinning.
```
Comment (michaelb42) [inline README.markdown]: The add-hardware script calls make menuconfig if no .config exists, so it's pretty self-explanatory

**Comment by TheSmartGerman** [comment]
Well, it didn’t worked for me. 

 

If I called the add-hardware script, it just opened a blank pinning.

 

After I went into the script itself, I saw there have to be an .config file.

 

So what a do for the moment:

-        I call menuconfig, choose my features, save to “Save Configuration to an Alternate File”, make sure naming is *.config

-        Then I call ./scripts/add_hardware user_my_hardware

 

Only if I do it like this, I will be able to assign the pins to the templates.

 

Right at the moment I’m just staring with ethersex, well slowly I’m getting used to it, but documentation is quite rare or hard to find.

 

So any help for beginning  would be great. 

 

Von: Michael Brakemeier [mailto:notifications@github.com] 
Gesendet: Donnerstag, 17. November 2016 21:09
An: ethersex/ethersex
Cc: TheSmartGerman; Author
Betreff: Re: [ethersex/ethersex] Explain adding hardware / make targets (#477)

 

@michaelb42 commented on this pull request.

  _____  

In README.markdown <https://github.com/ethersex/ethersex/pull/477#pullrequestreview-9107603> :

> @@ -14,7 +14,9 @@ The final hex file is named `ethersex.hex`.
 
 How to add a new hardware pinning
 =================================
-Use the script at `scripts/add-hardware` to add a new pinning.
+* Make `make menuconfig` and configure hardware / select features
+* Save configuration as .config file
+* Use the script at `scripts/add-hardware` to add a new pinning.

The add-hardware script calls make menuconfig if no .config exists, so it's pretty self-explanatory

**Comment by michaelb42** [comment]
Hm, works like a charm here on a fresh clone.

But anyway. the script add-hardware parses the .config file and provides templates for the pinning of the active i/o modules. You may also call it with a pre-exisiting .config, but it will not work with an alternate file, even if the file has the suffix .config. The argument to add-hardware will be prefixed with user_ by the script, you may but you probably should not provide it. The argument is solely used to generate your project specific hardware description as pinning/hardware/user_MYARGUMENT.m4.

Documentation can be found at http://www.ethersex.de/ and http://old.ethersex.de/. And you might also give IRC a try, you're welcome at #ethersex on freenode.

**Comment by TheSmartGerman** [comment]
well, ok. im still not get it sry. 

So if a call add-hardware, it will check for the latest changes in menuconfig?
After make mrproper, now add-hardware will also launch menuconfig first...

I think this is/was my problem. If you enter menuconfig once/and stored there will always a config file!?

**Comment by michaelb42** [comment]
Yes, exactly. If you want to start a new project it is a good idea to do a fresh clone or make mrproper in a new branch first.

But this is not meant to be a channel for help. btt plz. There are still open changes requested for this PR.

**Comment by michaelb42** [comment]
Suggest to close thos one due to lack of interest.

**Pull request closed without merging**